server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -431,20 +433,33 @@ public ClusterState execute(ClusterState currentState) {
                                 .getMaxNodeVersion()
                                 .minimumIndexCompatibilityVersion();
                             for (Map.Entry<String, String> indexEntry : indices.entrySet()) {
+                                String renamedIndexName = indexEntry.getKey();
                                 String index = indexEntry.getValue();
                                 boolean partial = checkPartial(index);
-                                SnapshotRecoverySource recoverySource = new SnapshotRecoverySource(
+
+                                IndexMetadata snapshotIndexMetadata = updateIndexSettings(
+                                    metadata.index(index),
+                                    request.indexSettings(),
+                                    request.ignoreIndexSettings()
+                                );
+                                final boolean isSearchableSnapshot;
+                                if (FeatureFlags.isEnabled(FeatureFlags.SEARCHABLE_SNAPSHOT)
+                                    && IndexModule.Type.REMOTE_SNAPSHOT.getSettingsKey().equals(request.storageType())) {
+                                    snapshotIndexMetadata = addSnapshotToIndexSettings(
+                                        snapshotIndexMetadata,
+                                        snapshot,
+                                        repositoryData.resolveIndexId(index)
+                                    );
+                                    isSearchableSnapshot = true;
+                                } else {
+                                    isSearchableSnapshot = false;
+                                }
+                                final SnapshotRecoverySource recoverySource = new SnapshotRecoverySource(
```

Comment:
Just to explain the change here...

The previous revision used an EmptyStoreRecoverySource here. It turns out that caused problems by not triggering completion of the "restore in progress" in the cluster state. I've instead added a field to SnapshotRecoverySource and branch off of that in the restore process. The net result is a bit simpler...I no longer have to change the method signatures in RoutingTable and IndexRoutingTable.

I discovered the bug by adding "wait_for_complete" to the snapshot restore request, which then never returned because the cluster state change never happened that would trigger the response from the server. The integ test now specifies this option.